packages/telemetry/src/sendTelemetry.ts

```diff
@@ -0,0 +1,224 @@
+export const sanitizeArgv = (argv: Array<string>) => {
+  const args = argv.slice(2)
+  const name = args[0]
+  const sensativeCommand =
+    SENSITIVE_ARG_POSITIONS[name as keyof SensitiveArgPositions]
+
+  if (sensativeCommand) {
+    // redact positional arguments
+    if (sensativeCommand.positions) {
+      sensativeCommand.positions.forEach((pos: number, index: number) => {
+        // only redact if the text in the given position is not a --flag
+        if (args[pos] && !args[pos].match(/--/)) {
+          args[pos] = sensativeCommand.redactWith[index]
+        }
+      })
+    }
+
+    // redact --option arguments
+    if (sensativeCommand.options) {
+      sensativeCommand.options.forEach((option: string, index: number) => {
+        const argIndex = args.indexOf(option)
+        if (argIndex !== -1) {
+          args[argIndex + 1] = sensativeCommand.redactWith[index]
+        }
+      })
+    }
+  }
+
+  return args.join(' ')
+}
+
+const buildPayload = async () => {
+  let payload: Record<string, unknown> = {}
+  let project
+
+  const argv = require('yargs/yargs')(process.argv.slice(2)).argv
+  const rootDir = argv.root
+  payload = {
+    type: argv.type || 'command',
+    command: argv.argv ? sanitizeArgv(JSON.parse(argv.argv)) : '',
+    duration: argv.duration ? parseInt(argv.duration) : null,
+    uid: uniqueId(rootDir),
+    ci: ci.isCI,
+    NODE_ENV: process.env.NODE_ENV || null,
+    ...(await getInfo({ redwoodVersion: argv.rwVersion })),
+  }
+
+  if (argv.error) {
+    payload.type = 'error'
+    payload.error = argv.error
+      .split('\n')[0]
+      .replace(/(\/[@\-\.\w]+)/g, '[path]')
+  }
+
+  // if a root directory was specified, use that to look up framework stats
+  // with the `structure` package
+  if (rootDir) {
+    project = new RWProject({
+      projectRoot: rootDir,
+      host: new DefaultHost(),
+    })
+  }
+
+  // add in app stats
+  payload = {
+    ...payload,
+    complexity: `${project.getRouter().routes.length}.${
+      project.services.length
+    }.${project.cells.length}.${project.pages.length}`,
+    sides: project.sides.join(','),
+  }
+
+  return payload
+}
+
+// returns the UUID for this device. caches the UUID for 24 hours
+const uniqueId = (rootDir: string | null) => {
+  const telemetryCachePath = path.join(
+    rootDir || '/tmp',
+    '.redwood',
+    'telemetry.txt'
+  )
+  const now = Date.now()
+  const expires = now - 24 * 60 * 60 * 1000 // one day
+  let uuid
+
+  if (
+    !fs.existsSync(telemetryCachePath) ||
+    fs.statSync(telemetryCachePath).mtimeMs < expires
+  ) {
+    uuid = uuidv4()
+    fs.writeFileSync(telemetryCachePath, uuid)
+  } else {
+    uuid = fs.readFileSync(telemetryCachePath).toString()
+  }
+
+  return uuid
+}
+
+// actually call the API with telemetry data
+export const sendTelemetry = async () => {
+  const telemetryUrl =
+    'https://telemetry.redwoodjs.com/.netlify/functions/telemetry'
```

Comment:
I would totally setup a proxy for this one and avoid us locking our selves in to telemetry

maybe a redirect like:
`'https://telemetry.redwoodjs.com/telemetry/push` ?

Reply:
Where does that live though, also Netlify? And the proxy isn't going to screw anything else up with headers or hostname or something? I feel like I always get caught in the 5% of cases where a proxy doesn't work exactly like you expect and I regret it forever.

Reply:
**For DNS** i.e. the subdomain
I personally would use cloudflare to manage all DNS, but thats me!

**For redirect**
Netlify if that's where you're deploying the telemetry function. I've moved my project between netlify, render and vercel a few times, all I did for netlify was:

```toml
#netlify.toml
[[redirects]]
  from = "/api/*"
  to = "/.netlify/functions/"
```

For render their dashboard, and for vercel that's the default one but can also be configured.

Reply:
Okay updated and tested a rewrite for Netlify! https://telemetry.redwoodjs.com/api/v1/telemetry